app/bundles/CoreBundle/Tests/Functional/DependencyInjection/Compiler/SystemThemeTemplatePathPassTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Tests\Functional\DependencyInjection\Compiler;
+
+use Mautic\CoreBundle\Helper\CacheHelper;
+use Mautic\CoreBundle\Helper\PathsHelper;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Request;
+
+class SystemThemeTemplatePathPassTest extends MauticMysqlTestCase
+{
+    protected function setUp(): void
+    {
+        // Create a twig file
+        $this->createOverrideFile();
+
+        // Clear the cache
+        /** @var CacheHelper $cacheHelper */
+        $cacheHelper = self::getContainer()->get('mautic.helper.cache');
+        $cacheHelper->nukeCache();
+
+        parent::setUp();
+    }
+
+    public function testUserProfilePageOverrideFromSystemThemDirectory(): void
+    {
+        $this->client->request(Request::METHOD_GET, '/s/account');
+
+        Assert::assertTrue($this->client->getResponse()->isOk());
+        Assert::assertStringContainsString('Override test', $this->client->getResponse()->getContent(), 'Page has not override.');
+    }
+
+    protected function beforeTearDown(): void
+    {
+        $path       = $this->getOverridePath();
+        $fileSystem = new Filesystem();
+        if ($fileSystem->exists($path)) {
+            $fileSystem->remove($path);
+        }
+    }
```

Comment:
I worry that this test could affect others. Shouldn't we nuke the cache here too?

Reply:
There is no need to nuke the cache. The twig engine ensures the template path is available; if not, it picks up another one. 

The following is the `php bin/console debug:twig` output,
<img width="752" alt="Screenshot 2024-04-05 at 3 15 52 PM" src="https://github.com/mautic/mautic/assets/1046788/7b4535cf-8921-4652-80aa-1436e90f0b3e">

So if there is nothing under the `themes/system/UserBundle/Resources/views/` path, then it's looking into `app/bundles/UserBundle/Resources/views/` as a fallback.